src/components/ui/text-field.svelte

```diff
@@ -38,29 +38,44 @@
       {name}
       bind:value
       cols={5}
-      on:focus={() => dispatch('focus')}
-      on:blur={() => dispatch('blur')}
-      on:input={() => dispatch('input')}
+      on:focus={(e) => dispatch('focus', e.target.value)}
+      on:blur={(e) => dispatch('blur', e.target.value)}
+      on:input={(e) => dispatch('input', e.target.value)}
       on:change={(e) => dispatch('change', e.target.value)}
     />
   {:else}
-    <input
-      {id}
-      type="text"
-      {placeholder}
-      {name}
-      {type}
-      bind:value
-      {pattern}
-      {min}
-      {max}
-      {maxLength}
-      on:focus={() => dispatch('focus')}
-      on:blur={() => dispatch('blur')}
-      on:input={() => dispatch('input')}
-      on:keyup.enter={() => dispatch('blur')}
-      on:change={(e) => dispatch('change', e.target.value)}
-    />
+    {#if isNum}
```

Comment:
Why the repetition? Can't type just depend on "isNum" and the other props have default value `null`?

(https://stackoverflow.com/questions/57392773/error-type-attribute-cannot-be-dynamic-if-input-uses-two-way-binding)

Reply:
Right. Managed doing sp